File: resources/[local]/rpc/server/sv_main.lua
local Functions, CallIdentifier = {}, {}, 0

RPC = {}

function RPC.register(name, func)
    Functions[name] = func
end

function RPC.remove(name)
    Functions[name] = nil
end

function ParamPacker(...)
    local params, pack = {...} , {}

    for i = 1, 15, 1 do
        pack[i] = {param = params[i]}
    end

    return pack
end

function ParamUnpacker(params, index)
    local idx = index or 1

    if idx <= #params then
        return params[idx]["param"], ParamUnpacker(params, idx + 1)
    end
end

function UnPacker(params, index)
    local idx = index or 1

    if idx <= 15 then
        return params[idx], UnPacker(params, idx + 1)
    end
end

RegisterNetEvent("rpc:request")
AddEventHandler("rpc:request", function(name, callID, params)
    local response
    local source = source

    if Functions[name] == nil then return end

    local success, error = pcall(function()
        response = ParamPacker(Functions[name](source, ParamUnpacker(params)))
    end)

    if not success then
        print ("Error: "..error)
        -- TriggerServerEvent("rpc:client:error", origin, name, error)
    end

    if response == nil then
        response = {}
    end

    TriggerClientEvent("rpc:response", source, callID, response)
end)

File: resources/[local]/rpc/server/sv_main.js
let Functions = {};
let RPC = {};

RPC.register = (name, func) => {
  Functions[name] = func;
};

RPC.remove = (name) => {
  Functions[name] = undefined;
};

const Packer = (params) => {
  let pack = [];
  for (let i = 0; i < 15; i++) {
    pack[i] = { param: params[i] };
  }
  return pack;
};

const UnPacker = (params) => {
  let shit = [];
  for (let i = 0; i < params.length; i++) {
    shit.push(params[i].param);
  }
  return shit;
};

RegisterNetEvent("rpc:request");
onNet("rpc:request", (resource, name, callID, params) => {
  let response;
  if (Functions[name] === undefined) return;
  try {
    response = Packer(Functions[name](source, UnPacker(params)));
  } catch {
    console.log("Error: " + error);
    return;
  }
  if (response === undefined) {
    response = {};
  }
  console.log("returning response: " + JSON.stringify(response));
  emitNet("rpc:response", source, resource, callID, response);
});

RPC.register("yess", () => {
  return ["yeah", undefined, "xdxdds"];
});
